### ISSUE TITLE
Q: How to determine which taxa to include?

### ISSUE THREAD

**Issue by nickdos** [AtlasOfLivingAustralia/arga#3]
Using NCBI as an example, how should we select taxa from the NCBI taxonomy to include in ARGA? Some possible approaches are:
- use all of the ALA taxonomy
- generate a list of taxa from ALA for any name with occurrence records found inside the Australian EEZ
- use name sources for known Australian species (AFD, APNI, APC, FungiMap) 

Inclusion of agricultural taxa may be an issues, as the ALA taxonomy may not cover these. So we may need to obtain a source for this - possibly from the DArT group.

**Comment by KathrynHallARGA** [comment]
We should understand how much undescribed material is on NCBI - we should compare to ALA list.

**Comment by KathrynHallARGA** [comment]
Is there a way to make our own from ALA + others we add?

**Comment by nickdos** [comment]
> Is there a way to make our own from ALA + others we add?

Yes, this is the plan. I've talked to Peggy and she's keen that the NCBI taxonomy is added as an additional source for ALA anyway, so we might get that for free.

**Comment by nickdos** [comment]
> We should understand how much undescribed material is on NCBI - we should compare to ALA list.

Not sure what you mean by "undescribed". Do you mean it doesn't have a scientific name, like an [OTU](https://en.wikipedia.org/wiki/Operational_taxonomic_unit)? Or are you referring to samples where is it identified to a higher taxa like genus or family?

**Comment by KathrynHallARGA** [comment]
I meant groups for which there is no formal species name.  These may be identified at species level, but with no name yet for some reason, or it may be a definable group of similar individuals, or even a singleton, that can't be classified further than family.  There is likely to be a mix of OTUs ranging from molecular species with a genus name, or maybe just a family, or perhaps even some "c.f.", "s.s.", "s.l.", or other taxonomic qualifiers like that.  I expect there is a jumble of levels from the hierarchy, depending on how well known the group is.

In practice it would be sort of the same - we could only have a higher taxon as "Thing-idae 1", "Thing-idae 2", but equally "Thingus A", etc., all of which would be effectively operational units.  It just depends on how far they were able to classify it when they gave the group an identifier.  

There is potential for this to balloon because some of them will likely be able to be combined or synonymised with more study...but it is not for us to make that judgement....but if that should happen by subsequent researchers, it would be advantageous that ARGA could track that history - so we would need to have those original names indexed for that purpose.  Similarly, difficulty arises when someone's Thingidae 1 is different to someone else's Thingidae 1, which could feasibly happen, so we would need to be very clear about carrying through that provenance to know who's Thingidae 1 is who's, and to which specimen/sequence the name belongs.

**Comment by nickdos** [comment]
The ALA name matching service attempts to find a best match for many of these incomplete "names". So if the record has some identifiable portion of the name (like genus) but the specific epithet is `sp.` or similar, then the record is identified at the genus level only. Same applies if it has `family sp` or higher. Also some records come with a code as the name (OTU, etc) but also provides a higher taxa additional property, like `family: Murdidae` or `kingdom: Bacteria`, then that name is used. Its only in cases where there is no identifiable portion of the name at all, that the record will have an empty scientific name.

In terms of finding records in (say) Genbank that fall into those various buckets I mentioned above, we shuolve be able to put together some counts of records that have `exact` matches versus the other incomplete match categories and lastly records with no name at all.

Here are some counts for the 2 Genbank Genome datasets:

```json
 { 
  "total records": "1,599,418",
  "unmatched records": "227,801",
  "matchType": [
    "exactMatch": "1,004,849",
    "higherMatch": "324,809",
    "canonicalMatch": "41,688",
    "fuzzyMatch": "269",
    "taxonIdMatch": "2",
    "phraseMatch": "0"
  ]
}
```

Here are records with names we couldn't match in Genbank: https://nectar-arga-dev-1.ala.org.au/?q=raw_scientificName:*%20AND%20-scientificName:*

**Comment by KathrynHallARGA** [comment]
That's fabulous!
OK, great.  I think that will get around a lot of the problem.  I imagine it is then also just a matter of making sure that we can track synonyms for the ones that match only to a higher taxon, in the event that their taxonomy should become better resolved in the future.  And also some mechanism for noting the taxonomic history for synonyms and homonyms too, like if My sp. A is the same name as Your sp. A, but is not the same organism.  (Or if My sp.A = My sp. C = Your sp. D).
